kyo-stm/shared/src/main/scala/kyo/STM.scala

```diff
@@ -145,4 +211,40 @@ object STM:
         }
 
     end run
+
+    private def quickSort(array: Array[Any], size: Int): Unit =
```

Comment:
Based on the implementation for `Async.gather` but specialized to the format of the flat ref log.